src/pages/filter/ui/panels/function-panel/components/inheritance-mode/inheritance-mode.tsx

```diff
@@ -53,21 +98,18 @@ export const InheritanceMode = observer(() => {
         handleChangeProblemGroups={handleChangeProblemGroups}
       />
 
-      <div className="h-px w-full bg-white my-4" />
+      <div className="h-px w-full bg-white my-3" />
 
       <ComplexVariants
-        variantsValues={variantsValues}
-        problemGroupValues={problemGroupValues}
-        filteredComplexVariants={filteredComplexVariants}
+        variantValues={variantValues}
+        variants={selectedFilter ? complexVariants : filteredComplexVariants}
         handleChangeVariants={handleChangeVariants}
       />
 
       <PanelButtons
         onSubmit={inheritanceModeStore.handleSumbitCondtions}
-        resetFields={() =>
-          inheritanceModeStore.handleResetAllFieldsLocally(problemGroupValues)
-        }
-        disabled={variantsValues.length === 0}
+        resetFields={() => inheritanceModeStore.resetAllFields()}
```

Comment:
{inheritanceModeStore.resetAllField}

Reply:
Does not work

Reply:
convert store methods to arrow functions.
the point is to avoid creating anonymous functions inside = {} on each component update. The more we avoid it - the better.